lib/worldwide/address.rb

```diff
@@ -463,5 +475,36 @@ def concatenation_delimiter(concatenated_field, decorator, field_index)
         concatenated_field.blank? ? RESERVED_DELIMITER : "#{decorator}#{RESERVED_DELIMITER}"
       end
     end
+
+    def parse_fields(field_definitions, field_values)
+      parsed_fields_hash = {}
+
+      field_definitions.each_with_index do |field, i|
+        field_name = field["key"].underscore
+        field_value = ""
+
+        if i < field_values.length
+          next_field = field_definitions[i + 1] unless last_field_in_arr?(field_definitions, i)
+          raw_field_value = field_values[i]
+          field_value = strip_decorators(next_field, raw_field_value)
+        end
+
+        parsed_fields_hash[field_name] = field_value
+      end
+      parsed_fields_hash
+    end
```

Comment:
I feel there may be a way to simplify this for better readability: 
```suggestion
def parse_fields(field_definitions, field_values)
  field_definitions.each_with_index.with_object({}) do |(field, i), parsed_fields|
    field_name = field["key"].underscore

    field_value = if i < field_values.length
                    raw_field_value = field_values[i]
                    next_field = field_definitions[i + 1] unless last_field_in_arr?(field_definitions, i)
                    strip_decorators(next_field, raw_field_value)
                  else
                    ""
                  end

    parsed_fields[field_name] = field_value
  end
end
```
What do you think?

Reply:
Oh nice, I like it! I forgot about `each_with_object`